### PR TITLE
GH-39072: [Release][CI] Python3.11-devel is required for the verification job on AlmaLinux 8

### LINKED ISSUE #39072: [Release][CI] Verification job on AlmaLinux 8 fails to find correct Python
### Describe the bug, including details regarding any error messages, version, and platform.

Arrow fails to be built on AlmaLinux 8 on the verification job due to not being able to find Python:
```
-- Build output directory: /tmp/arrow-14.0.2.ZwfXA/apache-arrow-14.0.2/python/build/temp.linux-x86_64-cpython-311/release
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Python3 (missing: Python3_INCLUDE_DIRS Development.Module
  NumPy) (found version "3.11.5")
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake/Modules/FindPython/Support.cmake:3165 (find_package_handle_standard_args)
  /usr/share/cmake/Modules/FindPython3.cmake:485 (include)
  cmake_modules/FindPython3Alt.cmake:51 (find_package)
  CMakeLists.txt:262 (find_package)


-- Configuring incomplete, errors occurred!
See also "/tmp/arrow-14.0.2.ZwfXA/apache-arrow-14.0.2/python/build/temp.linux-x86_64-cpython-311/CMakeFiles/CMakeOutput.log".
error: command '/usr/bin/cmake' failed with exit code 1
Failed to verify release candidate. See /tmp/arrow-14.0.2.ZwfXA for details.
1
```
AlmaLinux docker image was updated from 8.8 to 8.9 and the Python version there is different as found here: https://github.com/apache/arrow/pull/38985
The updated AlmaLinux Python version seems to be causing the issue.

This is reproducible on 14.0.2 RC0 and main:
[verify-rc-source-python-linux-almalinux-8-amd64](https://github.com/ursacomputing/crossbow/actions/runs/7086832209/job/19285845841)

### Component(s)

Continuous Integration, Release

### PR DESCRIPTION
**Pull request by raulcd**
### Rationale for this change

The verification task for Almalinux 8 was failing.

### What changes are included in this PR?

Add required python3.11-devel to the Docker image.

### Are these changes tested?

Yes via archery task.

### Are there any user-facing changes?

No

* Closes: #39072

### CODE REVIEW

**Comment by raulcd** [comment]
@github-actions crossbow submit verify-rc-source-python-linux-almalinux-8-amd64

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-manylinux-2-28-cp38-amd64

**Comment by raulcd** [comment]
@github-actions crossbow submit verify-rc-source-python-linux-almalinux-8-amd64

**Comment by raulcd** [comment]
@github-actions crossbow submit verify-rc-source-python-linux-almalinux-8-amd64

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-manylinux-2-28-cp38-amd64

**Review by pitrou — APPROVED** [review]

**Comment by raulcd** [comment]
@github-actions crossbow submit verify-rc-source-python-linux-almalinux-8-amd64

File: dev/release/setup-rhel-rebuilds.sh [hunk 1]
```diff
@@ -44,6 +44,7 @@ dnf -y install \
   ninja-build \
   nodejs \
   openssl-devel \
+  python3.11-devel \
```
[the remark this answers is not in the document]
Comment (raulcd) [reply dev/release/setup-rhel-rebuilds.sh]: I don't think it exists based on my packages search but I am trying it. If the job fails I'll revert the change

[the remark this answers is not in the document]
Comment (raulcd) [reply dev/release/setup-rhel-rebuilds.sh]: `python3-devel` seems to be `3.6.8` and it fails as [seen here](https://github.com/ursacomputing/crossbow/actions/runs/7104431351/job/19339357695#step:6:2169).

**Comment by raulcd** [comment]
@github-actions crossbow submit verify-rc-source-python-linux-almalinux-8-amd64

**Review by kou — APPROVED** [review]
+1

**Pull request merged**